File: internhub-frontend/src/models/PagedList.js
export class PagedList {
  constructor({
    currentPage = 1,
    lastPage = 1,
    databaseRecordsCount = 0,
    pageSize = 0,
    listSize = 0,
    data = [],
  }) {
    this.currentPage = currentPage;
    this.lastPage = lastPage;
    this.databaseRecordsCount = databaseRecordsCount;
    this.pageSize = pageSize;
    this.listSize = listSize;
    this.data = data;
  }

  static fromJson(json, dataList) {
    return new PagedList({
      currentPage: json["CurrentPage"],
      lastPage: json["LastPage"],
      databaseRecordsCount: json["DatabaseRecordsCount"],
      pageSize: json["PageSize"],
      listSize: json["ListSize"],
      data: dataList,
    });
  }
}

File: InternHub/InternHub.Common/PagedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InternHub.Common
{
    public class PagedList<T>
    {
        public int CurrentPage { get; set; } = 1;
        public int LastPage { get; set; } = 1;
        public int DatabaseRecordsCount { get; set; } = 0;
        public int PageSize { get; set; } = 0;
        public int ListSize { get => Data.Count; }
        public List<T> Data { get; set; } = new List<T>();
    }
}
